silhouette/app/com/mohiva/play/silhouette/impl/providers/OAuth2Provider.scala

```diff
@@ -165,6 +216,34 @@ trait OAuth2Provider extends SocialProvider with OAuth2Constants with Logger {
       info => Success(info)
     )
   }
+
+  def maybeRefresh(request: WSRequest, bearer: OAuth2Bearer): Future[(WSResponse, OAuth2Bearer)] = {
```

Comment:
val request = httpLayer.url("some/url/on/the/api").withMethod("post")
    val bearer = <from somewhere>
    val provider = <from somewhere>
    provider.maybeRefresh(request, bearer).map({ case (response, newBearer) =>
          dao.update(newBearer)
    })